persistent/Database/Persist/Types/Base.hs

```diff
@@ -293,6 +293,7 @@ data PersistValue = PersistText Text
                   | PersistList [PersistValue]
                   | PersistMap [(Text, PersistValue)]
                   | PersistObjectId ByteString -- ^ Intended especially for MongoDB backend
+                  | PersistArray [PersistValue] -- ^ Intended especially for PostgreSQL backend for text arrays
```

Comment:
This addition makes me a little uneasy. I don't think the `PersistValue` type should have too many values that are unsupported by the majority of backends.

Reply:
I think I'd rather have `PersistList` use the underlying Array type instead of adding a new type that throws a bunch of errors. Or we could have the `PersistArray` constructor fail over to the `PersistList` constructor.

Reply:
This might break people's programs, though. If suddenly `persistent` wants to read arrays, instead of the JSON-y implementation. Maybe a different way of handling this would be consolidating `PersistObjectId` and `PersistArray` into an `PersistBackendSpecific ByteString`, which can be used to input any kind of raw SQL?

Reply:
Yeah, altering the underlying datatype is definitely going to be a *big* breaking change for that. For now, I'm happy with treating this as `PersistList` in the non-Postgres backends and adding array functionality for this to Postgres.

Reply:
done